php/example_code/s3/S3Service.php

```diff
@@ -0,0 +1,238 @@
+        while (true) {
+            $listObjectsResult = $this->listObjects($bucketName, $start);
+            $contents = array_merge($contents, $listObjectsResult['Contents']);
+            if (!$listObjectsResult['IsTruncated']) {
+                break;
+            }
+            $start = $listObjectsResult['NextMarker'];
+        }
+        return $contents;
+    }
+
+    /**
+     * @param string $bucketName
+     * @param array $objects
+     * @param array $args - Additional arguments to be sent to the client.
+     * @return void
+     * @throws AwsException
+     */
+    public function deleteObjects(string $bucketName, array $objects, array $args = [])
+    {
+        $listOfObjects = array_map(function ($object) {
+            return ['Key' => $object];
+        }, array_column($objects, 'Key'));
+
+        $parameters = array_merge(['Bucket' => $bucketName, 'Delete' => ['Objects' => $listOfObjects]], $args);
+        try {
+            $this->client->deleteObjects($parameters);
+            if ($this->verbose) {
+                echo "Deleted the list of objects from: $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to delete the list of objects from $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with object deletion before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function deleteBucket(string $bucketName, array $args = [])
+    {
+        $parameters = array_merge(['Bucket' => $bucketName], $args);
+        try {
+            $this->client->deleteBucket($parameters);
+            if ($this->verbose) {
+                echo "Deleted the bucket named: $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to delete $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with bucket deletion before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function deleteObject(string $bucketName, string $fileName, array $args = [])
+    {
+        $parameters = array_merge(['Bucket' => $bucketName, 'Key' => $fileName], $args);
+        try {
+            $this->client->deleteObject($parameters);
+            if ($this->verbose) {
+                echo "Deleted the object named: $fileName from $bucketName.\n";
+            }
+        } catch (AwsException $exception) {
+            if ($this->verbose) {
+                echo "Failed to delete $fileName from $bucketName with error: {$exception->getMessage()}\n";
+                echo "Please fix error with object deletion before continuing.";
+            }
+            throw $exception;
+        }
+    }
+
+    public function listBuckets(array $args = [])
+    {
+        try {
+            $buckets = $this->client->listBuckets($args);
```

Comment:
Paginator?

Reply:
While listBuckets for PHP technically has a paginator, listBuckets is not actually paginated so it's not best use here.